src/wasm/wasm-binary.cpp

```diff
@@ -370,6 +371,43 @@ void WasmBinaryWriter::writeImports() {
   finishSection(start);
 }
 
+void WasmBinaryWriter::prepareFunctions() {
+  if (!options.generateStackIR) {
+    return;
+  }
+
+  // Generate StackIR for functions in parallel, and optimize if requested.
+  struct StackIRPass : public WalkerPass<PostWalker<StackIRPass>> {
+    bool isFunctionParallel() override { return true; }
+
+    std::unique_ptr<Pass> create() override {
+      return std::make_unique<StackIRPass>();
+    }
+
+    void doWalkFunction(Function* func) {
+      StackIRGenerator stackIRGen(*getModule(), func);
+      stackIRGen.write();
+      func->stackIR = std::make_unique<StackIR>();
+      func->stackIR->swap(stackIRGen.getStackIR());
```

Comment:
Does this work?
```suggestion
      func->stackIR = std::move(stackIRGen.getStackIR());
```

Reply:
```
src/wasm/wasm-binary.cpp:391:56: error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<std::vector<wasm::StackInst*> >’ and ‘std::remove_reference<std::vector<wasm::StackInst*>&>::type’ {aka ‘std::vector<wasm::StackInst*>’})
```

Reply:
Oh, then perhaps `func->stackIR = std::make_unique<StackIR>(std::move(stackIRGen.getStackIR()));`?